pydantic/utils.py

```diff
@@ -93,22 +93,63 @@ def almost_equal_floats(value_1: float, value_2: float, *, delta: float = 1e-8)
 class GetterDict:
```

Comment:
I don't see any way to set items (in case you wanted to replace the value of an existing key). But I'm assuming the point is that if you want that capability, you should subclass the `GetterDict`and put it in the model config. Works for me!

Reply:
Also, I think it's fine that it doesn't inherit from `Mapping` -- it may make it easier to test in the root_validator whether you got there by way of `from_orm` or `__init__` kwargs.

Reply:
Let me do some perf tests and think about this (if anyone else has an opinion I’d be interested)

Reply:
I thought about it a little more (but haven't perf tested). I'm hesitant to make any sacrifices to the perf of `from_orm` since I think it is an extremely commonly used function in fastapi, and in the vast majority of cases setting isn't necessary.

I could imagine shipping a `GetterSetterDict` that is like `GetterDict` but supports setting, but then requiring an override (similar to what I suggested in [this comment](https://github.com/samuelcolvin/pydantic/issues/380#issuecomment-535112498)) to support setting.

There might be an easier way to accomplish overriding (e.g. via config, but I don't want to pollute the config either..), but I would prefer an implementation that adds *zero* cost if not making use of the setting capability (like the override-based approach).

Reply:
Oh, I forgot you had already implemented the getter_dict via config. Yeah I think this is great, we should just also include a GetterSetterDict in the library for people to use if they need to override things. Would make it really easy to answer related issues.